openff/qcsubmit/results/caching.py

```diff
@@ -101,7 +104,15 @@ def _cached_client_query(
 
     client = _cached_fractal_client(client_address)
 
-    for batch_ids in _batched_indices(missing_query_ids, client.query_limit):
+    logger.debug(f"starting {query_name} to {client_address}")
+
+    batch_query_ids = _batched_indices(missing_query_ids, client.query_limit)
+    logger.debug(f"query split into {len(batch_query_ids)} batches")
+
+    for i, batch_ids in enumerate(batch_query_ids):
```

Comment:
A good place to use tqdm to show the progress of the queries.

Reply:
Definitely agree!